devtools/gradle/src/main/java/io/quarkus/gradle/dependency/ConditionalDependenciesEnabler.java

```diff
@@ -0,0 +1,193 @@
+                        validConditionalDependencies.add(extensionDependency);
+                        if (!extensionDependency.conditionalDependencies.isEmpty()) {
+                            featureVariants.putAll(extractFeatureVariants(Collections.singletonList(extensionDependency)));
+                        }
+                    }
+                }
+            }
+        }
+
+        Configuration enhancedDependencies = DependencyUtils.duplicateConfiguration(project,
+                project.getConfigurations().getByName(JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME));
+
+        List<ExtensionDependency> newConditionalDependencies = new ArrayList<>();
+
+        // The dependency may have been excluded, if no, no need to look for transitive dependencies
+        Set<ResolvedArtifact> resolvedArtifacts = enhancedDependencies.getResolvedConfiguration()
+                .getResolvedArtifacts();
+
+        for (ExtensionDependency extension : validConditionalDependencies) {
+            if (DependencyUtils.exists(resolvedArtifacts, extension.asDependency(dependencies))
+                    && !extension.conditionalDependencies.isEmpty()) {
+                newConditionalDependencies.add(extension);
+            }
+        }
+
+        if (!newConditionalDependencies.isEmpty()) {
+            resolveConditionalDependencies(project.getDependencies(),
+                    newConditionalDependencies, enhancedDependencies);
+        }
+    }
+
+    private Configuration createConditionalDependenciesConfiguration(Project project, Configuration existingDeps,
+            List<ExtensionDependency> extensions) {
+        Set<ResolvedArtifact> runtimeArtifacts = existingDeps.getResolvedConfiguration().getResolvedArtifacts();
+        List<Dependency> toResolve = new ArrayList<>();
+        for (Dependency dependency : collectConditionalDependencies(extensions)) {
+            if (!DependencyUtils.exists(runtimeArtifacts, dependency)) {
+                toResolve.add(dependency);
+            }
+        }
+        return project.getConfigurations()
+                .detachedConfiguration(toResolve.toArray(new Dependency[0]));
+    }
+
+    private Set<Dependency> collectConditionalDependencies(List<ExtensionDependency> extensionDependencies) {
+        Set<Dependency> dependencies = new HashSet<>();
+        for (ExtensionDependency extensionDependency : extensionDependencies) {
+            dependencies.addAll(extensionDependency.conditionalDependencies);
+        }
+        return dependencies;
+    }
+
+    private void enableConditionalDependency(ModuleVersionIdentifier dependency) {
+        ExtensionDependency extension = featureVariants.get(DependencyUtils.asFeatureName(dependency));
+        if (extension == null) {
+            return;
+        }
+        extension.importConditionalDependency(project, dependency);
+    }
+
+    private ExtensionDependency getExtensionInfoOrNull(ResolvedArtifact artifact, File artifactFile,
```

Comment:
It looks like the `artifactFile` is always `artifact.getFile()`, so the only method argument needed here is `artifact`.